File: 프로그래머스/lv0/120864. 숨어있는 숫자의 덧셈 （2）/숨어있는 숫자의 덧셈 （2）.java
class Solution {
    public int solution(String my_string) {
        int answer = 0;
		my_string = my_string.toLowerCase();
		my_string = my_string.replaceAll("[abcdefghijklmnopqrstuvwxyz]", "x");
		// my_string의 문자를 소문자, 알파벳을 x로 변환한다
		String[] arr = my_string.split("x");
        // 앞서 변환시킨 x를 기준으로 배열로 변환한다
        
		for(int i = 0; i < arr.length; i++) {
			if(!arr[i].isEmpty()) {				
				answer += Integer.parseInt(arr[i]);
			} // 배열의 원소가 비어있지 않다면 anwer에 숫자로 변환해서 누적합산한다
		}
        return answer;
    }
}

File: 프로그래머스/0/120864. 숨어있는 숫자의 덧셈 （2）/숨어있는 숫자의 덧셈 （2）.js
function solution(my_string) {
    let answer = 0;
    my_string = my_string.toLowerCase();
    my_string = my_string.replace(/[abcdefghijklmnopqrstuvwxyz]/g, "x");
    // 문자열의 소문자 알파벳을 모두 x로 대체한다
    let arr = my_string.split("x");
    // x를 기준으로 문자열을 분할하여 배열로 변환한다
    
    for (let i = 0; i < arr.length; i++) {
        if (arr[i] !== "") {
            answer += parseInt(arr[i]);
        } // 배열의 원소가 비어있지 않다면 숫자로 변환하여 누적합산한다
    }
    return answer;
}
